cmd/tuf/add_signatures.go

```diff
@@ -12,21 +12,34 @@ import (
 
 func init() {
 	register("add-signatures", cmdAddSignature, `
-usage: tuf add-signatures --signatures <sig_file> <metadata>
+usage: tuf add-signatures [--signatures <sig_file>] <metadata>
 
 Adds signatures (the output of "sign-payload") to the given role metadata file.
 
 If the signature does not verify, it will not be added.
+Options:
+  --signatures=<sig_file>   The path to the file containing the signatures to add. If not present, the contents are read from stdin
 `)
 }
 
 func cmdAddSignature(args *docopt.Args, repo *tuf.Repo) error {
 	roleFilename := args.String["<metadata>"]
 
-	f := args.String["<sig_file>"]
-	sigBytes, err := os.ReadFile(f)
-	if err != nil {
-		return err
+	f := args.String["--signatures"]
+	var sigBytes []byte
+	var err error
+	if f != "" {
+		sigBytes, err = os.ReadFile(f)
+		if err != nil {
+			return err
+		}
+	} else {
+		var input string
+		_, err := fmt.Scan(&input)
```

Comment:
[Scan()](https://pkg.go.dev/fmt#Scan) also returns a number of objects being scanned, so it'd be nice to take that into account and make sure you're getting exactly one object/signature.